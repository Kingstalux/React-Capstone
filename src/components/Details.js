import React from 'react';

export default function Details() {
  return (
    <div>
      <h2>Details</h2>
    </div>
  );
}
