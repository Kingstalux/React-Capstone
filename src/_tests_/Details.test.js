import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Details from '../components/Details';

describe('Details Element Tests', () => {
  test('Check if Component Renders in DOM', () => {
    render(
      <Provider store={store}>
        <Router>
          <Details />
        </Router>
      </Provider>,
    );
    expect(screen.queryByText(/2021/)).toBeNull();
    expect(screen.queryByText(/Description/)).toBeNull();
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Details />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
