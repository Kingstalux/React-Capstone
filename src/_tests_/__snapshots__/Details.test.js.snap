// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Element Tests renders correctly 1`] = `<div />`;
